src/ast/nodes/shared/FunctionNode.ts

```diff
@@ -91,6 +93,12 @@ export default class FunctionNode extends NodeBase {
 			this.isPrototypeReassigned = true;
 		}
 	}
+
+	render(code: MagicString, options: RenderOptions) {
+		this.context.inFunction = true;
```

Comment:
@lukastaegert nice catch - yes you're right, I've fixed up the state handling here.

The fact that you're saying the best solution in the paradigm you're advocating is not something that makes you feel happy indicates such a paradigm may not cater to the best patterns we need here.

Synchronous mutable state (aka finite state machines) are a common component of compilers, from tokenizing to visitors. Concepts of immutability while the current fad, are still beholden to the realities of computer science that avoiding memory copies and deep dynamic function calls will result in faster operation. Despite the hate, global state and visitor stack handling are useful constructs for the right tasks.

Certainly I'm all for implementing statelessness where we can, but, in short, code executing a visitor pattern should be allowed to have centralised stateful visitation, and this has been a proven way in compilers.

Reply:
It may be a proven compiler pattern but I think it is bad code style and definitely hurts maintainability to put a flag that describes the current state of a certain operation in a place where we only have flags that describe qualities of the whole module, graph or bundling process. 

If you must do it that way, please put it into an aptly named sub-state of the context like "renderState" to make it clear you want this information for a render operation so when in a few months from now we wonder what this weirdly unspecifically named `inFunction` is actually used for, we do not have to dig through the whole code base again and again to find out why it exists. Or anyone else who is not familiar with the codebase today.

In my personal opinion, maintainability should always be a top priority before considering some minor performance gains.